File: 0617.合并二叉树/0617-合并二叉树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* t1, TreeNode* t2) {
        if( !t1 && !t2 ) {
            return NULL;
        }

        int val = 0;
        if( t1 ) {
            val += t1->val;
        } 
        if( t2 ) {
            val += t2->val;
        }

        TreeNode* root = new TreeNode(val);
        root->left = mergeTrees( (t1 ? t1->left : NULL), (t2 ? t2->left : NULL) );
        root->right = mergeTrees( (t1 ? t1->right : NULL), (t2 ? t2->right : NULL) );
        return root;
    }
};

File: 0617.合并二叉树/0617-合并二叉树.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode mergeTrees(TreeNode t1, TreeNode t2) {
        if( t1 == null && t2 == null ) {
            return null;
        }
        
        int val = 0;
        if( t1 != null ) {
            val += t1.val;
        }
        if( t2 != null ) {
            val += t2.val;
        }

        TreeNode root = new TreeNode(val);
        root.left = mergeTrees( t1!=null ? t1.left:null,  t2!=null ? t2.left:null  );
        root.right = mergeTrees( t1!=null ? t1.right:null,  t2!=null ? t2.right:null  );
        return root;
    }
}